src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -445,17 +481,19 @@ function BarplotViz(props: VisualizationProps) {
       // barplot does not have independent axis range control so send undefined for defaultUIState
       truncationConfig(
         {
-          ...(minPos != null && max != null
-            ? {
-                dependentAxisRange: { min: minPos, max: max },
-              }
+          ...(dependentMinPosMax != null &&
+          dependentMinPosMax.min != null &&
+          dependentMinPosMax.max != null
+            ? { dependentAxisRange: dependentMinPosMax }
```

Comment:
I'll have another look at this tomorrow, but are you sure `truncationConfig` needs new 5th and 6th arguments?

If we pass `minPosMax` as the first argument and leave 5th and 6th undefined, I think it works fine.  Though I can't be sure. I still haven't understood what causes the bug where setting dependent range max to 0.5 in proportion mode erroneously causes the truncation warning as in #1350 (b) 

I'll look into #1350 (a) more too!

Reply:
@bobular Yes you are correct. Having those 5th and 6th arguments was actually my older approach before using minPosMax separately. I will change them accordingly.